### PR TITLE
feat: install plugin in STEAMPIPE_INSTALL_DIR if set, defaulting to ~/.steampipe. Also, let git ignore .steampipe directories

### PR DESCRIPTION
**Pull request by pdecat**
# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

**Comment by misraved** [comment]
@pdecat Is there a specific reason why we should add the `.steampipe` file extension to the `.gitignore` file?

Overall the change looks good, I believe this is something that we need to implement across all our plugin repos 👍 .

**Comment by pdecat** [comment]
> Is there a specific reason why we should add the .steampipe file extension to the .gitignore file?

I tend to set `STEAMPIPE_INSTALL_DIR` to `$(pwd)/.steampipe` during development and testing of plugins.
Ignoring that directory avoids mistakes.

**Comment by cbruno10** [comment]
@pdecat What's the reason you set your install dir to `$(pwd)/.steampipe`? From our development experience at least, that's usually not standar, but curious to know what your use case is.

**Comment by pdecat** [comment]
That's to experiment with the plugin in isolation, especially needed when setting `STEAMPIPE_PPROF=1`. For an example, see https://turbot-community.slack.com/archives/C01UECB59A7/p1690455022578639?thread_ts=1690018753.077079&cid=C01UECB59A7

**Comment by cbruno10** [comment]
From that thread, I remember the use case of needing to test in isolation, but I think that's not normally required for plugin development.

Although a bit harmless, I'm hesitant to add `.steampipe` to `.gitignore` since I believe the use case behind it is very rare (at least it hasn't come up while developing Turbot plugins or community plugins yet). If I'm missing anything else about its use though, please let me know.

You can still add it as part of `.gitignore` locally or not commit your `.steampipe` directory, but for this PR, can you please revert the change in `.gitignore`?

File: .gitignore [hunk 1]
```diff
@@ -34,4 +34,7 @@ scripts/generate_parliament_iam_permissions/generate_go_file.
 node_modules
 
 # Ignore file types
-.vscode/
\ No newline at end of file
+.vscode/
+
+# Ignore steampipe install directories
+.steampipe
```
Comment (pdecat) [inline .gitignore:40]:
```suggestion
```

**Comment by pdecat** [comment]
Good for me, done!

**Review by cbruno10 — APPROVED** [review]

**Comment by cbruno10** [comment]
Thanks @pdecat for the update!

**Pull request merged**